herd/AArch64ASLSem.ml

```diff
@@ -445,62 +473,30 @@ module Make (TopConf : AArch64Sig.Config) (V : Value.AArch64ASL) :
           in
           None
 
-    let symb_values_tbl : (string, V.v) Hashtbl.t = Hashtbl.create 17
-    let symb_values_tag = "AArch64-HashedValue"
+    let tr_cst tr =
+      Constant.map tr
+        (fun _ -> Warn.fatal "Cannot translate PTE")
+        (fun _ -> Warn.fatal "Cannot translate instruction")
+
+    let aarch64_to_asl_bv = function
+      | V.Var v -> ASLS.A.V.Var v
+      | V.Val cst ->
+         ASLS.A.V.Val
+           (tr_cst ASLScalar.convert_to_bv cst)
+
+    let aarch64_to_asl = function
+      | V.Var v -> ASLS.A.V.Var v
+      | V.Val cst ->
+         ASLS.A.V.Val (tr_cst Misc.identity cst)
+
+    let asl_to_aarch64 = function
+      | ASLS.A.V.Var v -> V.Var v
+      | ASLS.A.V.Val cst ->
+         V.Val (tr_cst Misc.identity cst)
 
     let fake_test ii fname decode =
-      let proc = 0 in
-      let init =
-        let loc r = MiscParser.Location_reg (proc, AArch64Base.pp_reg r) in
-        let tr_c =
-          let open Constant in
-          function
-          | Concrete s -> Concrete (V.Cst.Scalar.pp false s)
-          | Symbolic s -> Symbolic s
-          | c ->
-             let name = V.Cst.pp_v c and tag = Some symb_values_tag in
-             let () = Hashtbl.add symb_values_tbl name (V.Val c) in
-             Symbolic (Virtual { offset = 0; cap = 0L; tag; name }) in
-        let set r c =
-          let c' = tr_c c in
-          (loc r, (TestType.TyDef, c'))
-        in
-        let init_gregs =
-          let one_reg r =
-            match A.look_reg r ii.A.env.A.regs with
-            | Some (V.Val c) -> Some (set r c)
-            | Some (V.Var _) | None -> None
-          in
-          List.filter_map one_reg ASLBase.gregs
-        in
-        let nzcv = AArch64Base.NZCV in
-        let vnzcv =
-          match A.look_reg nzcv ii.A.env.A.regs with
-          | Some (V.Val c) -> Some (tr_c c)
-          | Some (V.Var _) -> None
-          | None ->
-             Some (Constant.Concrete "0") in
-        let () =
-          if _dbg then
-            let pp =
-              match vnzcv with
-              | Some v -> ParsedConstant.pp_v v
-              | None -> "-" in
-            Printf.eprintf "Initial value of flags: %s\n%!" pp
-        in
-        match vnzcv with
-        |Some c ->
-          let tc = TestType.TyDef,c and pstate = "PSTATE" in
-          (loc nzcv,tc)
-          (* The initial value for PSTATE given bellow
-           * overrides the one of the "patches.asl" file,
-           *  see `herd/ASLSem.ml`.
-           *)
-          ::(MiscParser.Location_reg (proc,pstate),tc)
-          ::init_gregs
-        | None -> init_gregs
-        in
-        let prog =
+      let init = [] in
```

Comment:
Why do we still have this?
And why did you move the construction of init after?

Reply:
Aha, If we construct init before calling `ASLTH.build`  we  have to turn AArch64ASL values into `ParsedConstant.v`'s. When we do it after, we translate AArch64ASL values into ASL values, which are basically the same!

Reply:
Oh smart!